blog/content/post/2020-08-26-coreboot-riscv.md

```diff
@@ -0,0 +1,267 @@
+```
+
+
+#### 7. Test image in QEMU
+
+Firstly you need to convert `coreboot.rom` to an ELF that Qemu can load
+```sh
+$ util/riscv/make-spike-elf.sh build/coreboot.rom build/coreboot.elf
+```
+
+Now you can run your image in Qemu
+
+```sh
+$ qemu-system-riscv64 -M virt -m 1024M -nographic -kernel build/coreboot.elf
+```
+
+You should see similar output
+
+```
+coreboot-4.12-2440-g2afee12991-dirty Fri Aug 28 06:07:37 UTC 2020 bootblock starting (log level: 7)...
+FMAP: Found "FLASH" version 1.1 at 0x20000.
+FMAP: base = 0x0 size = 0x400000 #areas = 4
+FMAP: area COREBOOT found @ 20200 (4062720 bytes)
+CBFS: Locating 'fallback/romstage'
+CBFS: Found @ offset 80 size 3733
+BS: bootblock times (exec / console): total (unknown) / 0 ms
+
+
+coreboot-4.12-2440-g2afee12991-dirty Fri Aug 28 06:07:37 UTC 2020 romstage starting (log level: 7)...
+RAMDETECT: Found 1020 MiB RAM
+CBMEM:
+IMD: root @ 0xbffff000 254 entries.
+IMD: root @ 0xbfffec00 62 entries.
+FMAP: area COREBOOT found @ 20200 (4062720 bytes)
+CBFS: Locating 'fallback/ramstage'
+CBFS: Found @ offset 3800 size 5ae8
+BS: romstage times (exec / console): total (unknown) / 0 ms
+
+
+coreboot-4.12-2440-g2afee12991-dirty Fri Aug 28 06:07:37 UTC 2020 ramstage starting (log level: 7)...
+Enumerating buses...
+RAMDETECT: Found 1020 MiB RAM
+CBMEM:
+IMD: root @ 0xbffff000 254 entries.
+IMD: root @ 0xbfffec00 62 entries.
+Root Device scanning...
+CPU_CLUSTER: 0 enabled
+scan_bus: bus Root Device finished in 0 msecs
+done
+Allocating resources...
+Reading resources...
+CPU_CLUSTER: 0 missing read_resources
+Done reading resources.
+Done setting resources.
+Done allocating resources.
+Enabling resources...
+done.
+Initializing devices...
+Devices initialized
+Finalize devices...
+Devices finalized
+Writing coreboot table at 0xbffdc000
+ 0. 0000000080400000-0000000080431fff: RAM
+ 1. 0000000080432000-0000000080446fff: RAMSTAGE
+ 2. 0000000080447000-0000000081431fff: RAM
+ 3. 0000000081432000-0000000081432fff: RAMSTAGE
+ 4. 0000000081433000-00000000bffdbfff: RAM
+ 5. 00000000bffdc000-00000000bfffffff: CONFIGURATION TABLES
+FMAP: area COREBOOT found @ 20200 (4062720 bytes)
+Wrote coreboot table at: 0xbffdc000, 0x170 bytes, checksum f993
+coreboot table: 392 bytes.
+IMD ROOT    0. 0xbffff000 0x00001000
+IMD SMALL   1. 0xbfffe000 0x00001000
+CONSOLE     2. 0xbffde000 0x00020000
+COREBOOT    3. 0xbffdc000 0x00002000
+IMD small region:
+  IMD ROOT    0. 0xbfffec00 0x00000400
+FMAP: area COREBOOT found @ 20200 (4062720 bytes)
+CBFS: Locating 'fallback/payload'
+CBFS: Found @ offset 9740 size 11187
+Checking segment from ROM address 0x80029978
+SELF segment doesn't target RAM: 0x000dfa80, 132480 bytes
+ 0. 0000000080400000-0000000080431fff: RAM
+ 1. 0000000080432000-0000000080446fff: RAMSTAGE
+ 2. 0000000080447000-0000000081431fff: RAM
+ 3. 0000000081432000-0000000081432fff: RAMSTAGE
+ 4. 0000000081433000-00000000bffdbfff: RAM
+ 5. 00000000bffdc000-00000000bfffffff: CONFIGURATION TABLES
```

Comment:
Is this only coreboot log or seabios as well?

Reply:
As said in previous comment, SeaBIOS is likely not supported on RISC-V. Try Linux payload or LinuxBoot